js/store.js

```diff
@@ -195,42 +195,85 @@ const store = {
 
     const website = await this.db.websites.get(hostname);
     if (website) {
-      return website;
+      const websiteOutput = {};
+      Object.keys(website).forEach((key) => {
+        websiteOutput[key] = this.mungeDataOutbound(key, website[key]);
+      });
+      return websiteOutput;
     } else {
       return {};
     }
   },
 
-  async setWebsite(hostname, data) {
+  mungeDataInbound(key, value) {
+    if (this.indexes.includes(key)) {
+      // IndexedDB does not accept boolean values for indexes; using 0/1 instead
+      if (value === true) {
+        value = 1;
+      }
+      if (value === false) {
+        value = 0;
+      }
+    }
+    return value;
+  },
+
+  mungeDataOutbound(key, value) {
+    if (this.indexes.includes(key)) {
+      // IndexedDB does not accept boolean values for indexes; using 0/1 instead
+      if (value === 1) {
+        value = true;
+      }
+      if (value === 0) {
+        value = false;
+      }
+    }
+    return value;
+  },
 
+  async setWebsite(hostname, data) {
     const website = await this.getWebsite(hostname);
 
     if (!('hostname' in website)) {
       website['hostname'] = hostname;
     }
 
     for (const key in data) {
-      // store first and last request times for clearing data every X days
-      if (key === 'requestTime') {
-        if (!('firstRequestTime' in data)) {
-          website['firstRequestTime'] = data[key];
-        } else {
-          website['lastRequestTime'] = data[key];
-        }
-      } else {
-        // IndexedDB does not accept boolean values; using 0/1 instead
-        if (data[key] === true) {
-          data[key] = 1;
-        }
-        if (data[key] === false) {
-          data[key] = 0;
-        }
-
-        website[key] = data[key];
+      const value = this.mungeDataInbound(key, data[key]);
+      switch (key) {
+        case 'requestTime':
+          // store first and last request times for clearing data every X days
+          if (!('firstRequestTime' in website)) {
+            website.firstRequestTime = value;
+          } else {
+            website.lastRequestTime = value;
+          }
+          break;
+        case 'isVisible':
+          if ('isVisible' in website
+              && website.isVisible === true) {
+            break;
+          }
+          website.isVisible = value;
+          break;
+        case 'firstParty':
+          if ('firstParty' in website
+              && website.firstParty === true) {
```

Comment:
This is saying we NEVER want a site that's already stored as a `firstParty` `true` to be switched to `false`.